CodeU_Assignement_5/src/codeu_assignement_5/KemUnknownLanguage.java

```diff
@@ -0,0 +1,145 @@
+package codeu_assignement_5;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ *
+ * @author Karine
+ */
+public class KemUnknownLanguage {
+    /*
+     * Input: a dictionary (a list of words in lexicographic order) of all words in an unknown/invented language, 
+    *  Output: alphabet (an ordered list of characters) of that language, letters we cannot decide are appended in the end.
+    */
+    public List<Character> getAlphabeit(String[] alienWords) {
+        if ((alienWords != null) && (alienWords.length > 0)) {
+           return extractAlphabeitFromWords(alienWords);
+        }
+        
+        return null;
+    }
+
+    private List<Character> extractAlphabeitFromWords(String[] alienWords) {
+        /* Get the current order by prefix size, starting from 0 */
+        List<Character> dict = getBasicAlphaBeit(alienWords, 0); /* Basic Alphabeit */ 
+        /* A,R,C from the example! */
+
+        /* Add from prefix while there are letters after first letter of prefix */
+        while (!addToDictByPrefix(alienWords, dict)){} 
+
+        /* Add all the chars we cannot decide in the end*/
+        for (String w : alienWords) {
+            for (Character c: w.toCharArray()) {
+                if (!dict.contains(c)) {
+                    dict.add(c);
+                }
+            }
+        }
+        
+        return dict;
+    }
```

Comment:
I do really try to learn what is the right level of comments. Before, I've said that "the code shall document itself", now, that there shall be comments describing the code.

It is for sure, will not help me to learn.

Regarding the bugs, I would expect some suggestion of cases (and not to base your answer on feelings). However, what I can do, is to add all other team members cases to see if it works well.